src/Extensions/Logging/Internal/ServiceInitializationEventSource.cs

```diff
@@ -23,47 +23,67 @@ internal sealed class ServiceInitializationEventSource : EventSource
 		/// </summary>
 		/// <param name="hostProcessId">Host process id</param>
 		/// <param name="serviceType">Service type</param>
+		/// <param name="message">Optional message</param>
 		[NonEvent]
-		public void LogHostBuildSucceeded(int hostProcessId, string serviceType)
+		public void LogHostBuildSucceeded(int hostProcessId, string serviceType, string message = "")
 		{
 			if (!IsEnabled())
 			{
 				return;
 			}
 
-			LogHostBuildSucceeded(
+			string logMessage = string.IsNullOrWhiteSpace(message) ?
+				FormattableString.Invariant($"Service host process {hostProcessId} registered service type {serviceType}") :
+				message;
+			LogHostBuildSucceededInternal(
 				hostProcessId,
 				serviceType,
-				FormattableString.Invariant($"Service host process {hostProcessId} registered service type {serviceType}"));
+				logMessage);
 		}
 
 		/// <summary>
 		/// Logs a generic host failure
 		/// </summary>
 		/// <param name="exception">Exception</param>
 		/// <param name="serviceType">Service type</param>
+		/// <param name="message">Optional message</param>
 		[NonEvent]
-		public void LogHostFailed(string exception, string serviceType)
+		public void LogHostFailed(string exception, string serviceType, string message = "")
 		{
 			if (!IsEnabled())
 			{
 				return;
 			}
+			string logMessage = string.IsNullOrWhiteSpace(message) ?
+				FormattableString.Invariant($"Service host initialization failed for {serviceType} with exception {exception}") :
+				message;
 
-			LogHostFailed(
+			LogHostFailedInternal(
 				exception,
 				serviceType,
-				FormattableString.Invariant($"Service host initialization failed for {serviceType} with exception {exception}"));
+				message);
 		}
 
 		private ServiceInitializationEventSource() { }
 
+		/// <summary>
+		/// Log host build succeeded
+		/// </summary>
+		/// <param name="hostProcessId">Host process id</param>
+		/// <param name="serviceType">The service type</param>
+		/// <param name="message">The message to be logged</param>
 		[Event((int)EventSourcesEventIds.GenericHostBuildSucceeded, Level = EventLevel.Informational, Message = "{2}", Version = 1)]
-		private void LogHostBuildSucceeded(int hostProcessId, string serviceType, string message) =>
+		private void LogHostBuildSucceededInternal(int hostProcessId, string serviceType, string message) =>
```

Comment:
I'm a bit worried about the need of remaining private methods, but as far as I know, it's not the part of the contract so it should be fine